### PR TITLE
Forced rotation:  Remove cached JWT-SVIDs using tainted keys

### PR DESCRIPTION
**Pull request by amartinezfayo**
Remove from the agent cache the JWT-SVIDs using tainted keys

Fixes: #3904.

### CODE REVIEW

File: pkg/agent/manager/sync.go [hunk 1]
```diff
@@ -76,11 +82,9 @@ func (m *manager) processTaintedAuthorities(ctx context.Context, bundle *spiffeb
 		}
 	}
 
-	newTaintedJWTAuthorities := getNewItems(m.processedTaintedJWTAuthorities, jwtAuthorities)
-	if len(newTaintedJWTAuthorities) > 0 {
-		m.c.Log.WithField(telemetry.SubjectKeyIDs, strings.Join(newTaintedJWTAuthorities, ",")).
-			Debug("New tainted JWT authorities found")
-		// TODO: IMPLEMENT!!!
+	if len(jwtAuthorities) > 0 {
```
Comment (MarcosDY) [inline pkg/agent/manager/sync.go:85]: may we compare new tainted authorities with already tainted authorities? 
if not we'll be creating logs and metrics for something that there must be empty, right?

File: pkg/common/telemetry/agent/manager.go [hunk 2]
```diff
@@ -46,10 +51,20 @@ func AddCacheManagerOutdatedSVIDsSample(m telemetry.Metrics, cacheType string, c
...
+	}
+	m.AddSample(key, count)
+}
+
+// AddCacheManagerTaintedJWTSVIDsSample count of tainted SVIDs according to
 // agent cache manager
-func AddCacheManagerTaintedSVIDsSample(m telemetry.Metrics, cacheType string, count float32) {
-	key := []string{telemetry.CacheManager, cacheType, telemetry.TaintedSVIDs}
+func AddCacheManagerTaintedJWTSVIDsSample(m telemetry.Metrics, cacheType string, count float32) {
+	key := []string{telemetry.CacheManager, telemetry.CountJWTSVIDs}
```
Comment (MarcosDY) [inline pkg/common/telemetry/agent/manager.go:67]: can ypu add a label that display it is a sample related with tainting?

File: pkg/agent/manager/sync.go [hunk 3]
```diff
@@ -76,11 +82,14 @@ func (m *manager) processTaintedAuthorities(ctx context.Context, bundle *spiffeb
...
 	}
 
-	newTaintedJWTAuthorities := getNewItems(m.processedTaintedJWTAuthorities, jwtAuthorities)
+	newTaintedJWTAuthorities := getNewItemsFromMap(m.processedTaintedJWTAuthorities, jwtAuthorities)
 	if len(newTaintedJWTAuthorities) > 0 {
-		m.c.Log.WithField(telemetry.SubjectKeyIDs, strings.Join(newTaintedJWTAuthorities, ",")).
-			Debug("New tainted JWT authorities found")
-		// TODO: IMPLEMENT!!!
+		// Taint JWT-SVIDs in the cache
+		m.cache.TaintJWTSVIDs(ctx, jwtAuthorities)
```
Comment (MarcosDY) [inline pkg/agent/manager/sync.go:88]: can you keep the log to display that new tainted keys found? that will simplify debugging  tainted authorities propagation

File: pkg/agent/manager/cache/lru_cache.go [hunk 4]
```diff
@@ -635,7 +636,7 @@ func (c *LRUCache) notifyTaintedBatchProcessed() {
 
 // processTaintedSVIDs identifies and removes tainted SVIDs from the cache that have been signed by the given tainted authorities.
 func (c *LRUCache) processTaintedSVIDs(entryIDs []string, taintedX509Authorities []*x509.Certificate) {
-	counter := telemetry.StartCall(c.metrics, telemetry.CacheManager, "", telemetry.ProcessTaintedSVIDs)
+	counter := telemetry.StartCall(c.metrics, telemetry.CacheManager, agent.CacheTypeWorkload, telemetry.ProcessTaintedX509SVIDs)
```
[the remark this answers is not in the document]
Comment (amartinezfayo) [reply pkg/agent/manager/cache/lru_cache.go:639]: Would be ok if I update the telemetry doc in a separate PR?

[the remark this answers is not in the document]
Comment (MarcosDY) [reply pkg/agent/manager/cache/lru_cache.go:639]: sure

**Review by MarcosDY — APPROVED** [review]

**Pull request merged**